src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,201 @@
+package duke;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.enums.CommandWord;
+import duke.enums.Message;
+import duke.exception.DukeException;
+
+import java.util.StringTokenizer;
+
+/**
+ * Parses the user input to determine what command the user intends to make
+ */
+public class Parser {
+    
+    private static final String DEADLINE_DELIMITER = "/by";
```

Comment:
I like how you declared final constants for the delimiters. I will learn from this for my own code.

Reply:
Credit goes to IntelliJ :laughing: